### PR TITLE
Minor refactor to eliminate code redundancy

### PR DESCRIPTION
**Pull request by jamesgeorge007**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
N/A

### Motivation / Use-Case
To eliminate redundanct code resulting in much concise code-base.

### Breaking Changes
N/A

### Additional Info
N/A

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
cc @hiroppy

File: lib/Server.js [hunk 1]
```diff
@@ -292,14 +290,12 @@ class Server {
...
     this.options.proxy.forEach((proxyConfigOrCallback) => {
-      let proxyConfig;
       let proxyMiddleware;
 
-      if (typeof proxyConfigOrCallback === 'function') {
-        proxyConfig = proxyConfigOrCallback();
-      } else {
-        proxyConfig = proxyConfigOrCallback;
-      }
+      let proxyConfig =
```
Comment (evilebottnawi) [inline lib/Server.js]: maybe const?

**Comment by jamesgeorge007** [comment]
cc @evilebottnawi

**Comment by Loonride** [comment]
@jamesgeorge007 Good idea, but this may be unnecessary since `serverMode` and `clientMode` will be removed soon in favor of a single `transportMode` option as explained here: https://github.com/webpack/webpack-dev-server/issues/1860#issuecomment-510133108. Still, it can't hurt to have this while these options are still around.

**Comment by jamesgeorge007** [comment]
@Loonride @evilebottnawi shall we land this one before #1860 :thinking:

**Comment by jamesgeorge007** [comment]
@Loonride @evilebottnawi shall we land this one before #1860 :thinking:

**Comment by hiroppy** [comment]
Yes, I think it is better.

**Comment by jamesgeorge007** [comment]
@hiroppy So are you gonna go ahead and merge this one? It seems like #1860 is all set to be merged.

**Comment by hiroppy** [comment]
Sorry, I misread it. I wanna wait for changing `transportMode` because it'll be merge shortly.

File: lib/Server.js [hunk 2]
```diff
@@ -153,6 +139,18 @@ class Server {
...
+    ['serverMode', 'clientMode'].forEach((mode) => {
+      if (this.options[mode] === undefined) {
+        this.options[mode] = 'sockjs';
+      } else {
+        this.log.warn(
+          `${mode} is an experimental option, meaning its usage could potentially change without warning`
+        );
+      }
+    });
+  }
```
Comment (jamesgeorge007) [inline lib/Server.js]: I guess I should revert these changes after the respective merge

**Comment by Loonride** [comment]
@jamesgeorge007 I suggest you remove the `serverMode/clientMode` changes since https://github.com/webpack/webpack-dev-server/pull/2116 will likely be merged shortly and there will be more changes to come for `transportMode` default setting. Keep the `proxyConfig` changes though.

File: lib/Server.js (hunk 1 above)
Comment (Loonride) [reply lib/Server.js]: Actually I think let is needed, since `proxyConfig` is set to something else later under certain conditions. Probably why tests are failing.

**Comment by jamesgeorge007** [comment]
cc @Loonride

**Comment by jamesgeorge007** [comment]
cc @evilebottnawi

**Comment by evilebottnawi** [comment]
Need fix problem with lint

**Comment by jamesgeorge007** [comment]
@evilebottnawi It seems that everything's fine now :thinking:

**Comment by hiroppy** [comment]
I don't know this change is needed.

**Comment by evilebottnawi** [comment]
Just less code :smile:

**Comment by jamesgeorge007** [comment]
It definitely makes sense inorder to be consistent with rest of the
codebase, right?
